spring-cloud-kubernetes-fabric8-config/src/main/java/org/springframework/cloud/kubernetes/fabric8/config/reload/Fabric8EventBasedConfigMapChangeDetector.java

```diff
@@ -46,57 +55,78 @@ public class Fabric8EventBasedConfigMapChangeDetector extends ConfigurationChang
 
 	private final boolean monitoringConfigMaps;
 
-	private SharedIndexInformer<ConfigMap> informer;
+	private final List<SharedIndexInformer<ConfigMap>> informers = new ArrayList<>();
+
+	private final Set<String> namespaces;
+
+	private final boolean enableReloadFiltering;
+
+	private final ResourceEventHandler<ConfigMap> handler = new ResourceEventHandler<>() {
+
+		@Override
+		public void onAdd(ConfigMap configMap) {
+			onEvent(configMap);
+		}
+
+		@Override
+		public void onUpdate(ConfigMap oldConfigMap, ConfigMap newConfigMap) {
+			onEvent(newConfigMap);
+		}
+
+		@Override
+		public void onDelete(ConfigMap configMap, boolean deletedFinalStateUnknown) {
+			onEvent(configMap);
+		}
+
+		// leave as comment on purpose, may be this will be useful in the future
```

Comment:
what is this needed for?

Reply:
I am leaving it here just in case someone after me (or I will forge too :) ) will face issues related to HTTP_GONE, even if fabric8 team said that this is not supposed to be the case... 

So the idea is that using an informer instead of a watcher, is supposed to get rid of HTTP_GONE problem at all. At the same time, fabric8 provides a `default` method that allows you to track those errors. 

So in the very slim case that things go wrong - we have a way to fix things. I doubt this is ever going to be needed, but just in case, I decided to leave it as a comment.

Reply:
What the harm in having the code there, even if it should never get hit?

Reply:
I am more afraid of the reverse - it might do more harm than good. From Marc's words, we should really override that only for tracking purposes, because the informer should heal itself, it's a `default` method for a reason. So my reasoning for leaving that code - is "just in case" I will be faced with any potential issues - to know were to start debugging from.

Reply:
darn that's a good point. on it.

Reply:
I'm going to take my time and see if/how we can have an integration test for this.